go/kbfs/libkbfs/folder_branch_ops.go

```diff
@@ -759,6 +759,120 @@ func (fbo *folderBranchOps) clearConflictView(ctx context.Context) (
 	return fbo.cr.clearConflictRecords()
 }
 
+// forceStuckConflictForTesting forces the TLF into a stuck conflict
+// view, for testing.
+func (fbo *folderBranchOps) forceStuckConflictForTesting(
+	ctx context.Context) (err error) {
+	startTime, timer := fbo.startOp(ctx, "Forcing a stuck conflict")
+	defer func() {
+		fbo.endOp(
+			ctx, startTime, timer, "Forcing a stuck conflict done: %+v", err)
+	}()
+
+	lState := makeFBOLockState()
+	fbo.mdWriterLock.Lock(lState)
+	defer fbo.mdWriterLock.Unlock(lState)
+
+	if fbo.isUnmergedLocked(lState) {
+		return errors.New("Cannot force conflict when already unmerged")
+	}
+
+	// Disable updates.
+	unpauseUpdatesCh := make(chan struct{})
+	fbo.updatePauseChan <- unpauseUpdatesCh
+	defer func() { unpauseUpdatesCh <- struct{}{} }()
+
+	// Make a no-op revision with an empty resolutionOp. Wait for it
+	// to flush to the server.
+	origHead, _ := fbo.getHead(ctx, lState, mdNoCommit)
+	mergedGCOp := newGCOp(origHead.data.LastGCRevision)
+	err = fbo.finalizeGCOpLocked(ctx, lState, mergedGCOp)
+	if err != nil {
+		return err
+	}
+
+	jManager, _ := GetJournalManager(fbo.config)
+	if jManager != nil {
+		err := fbo.waitForJournalLocked(ctx, lState, jManager)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Roll back the local view to the original revision.
+	err = func() error {
+		fbo.headLock.Lock(lState)
+		defer fbo.headLock.Unlock(lState)
+		err = fbo.setHeadLocked(ctx, lState, origHead, headTrusted, mdNoCommit)
+		if err != nil {
+			return err
+		}
+		fbo.setLatestMergedRevisionLocked(
+			ctx, lState, origHead.Revision(), true)
+		return nil
+	}()
+	if err != nil {
+		return err
+	}
+
+	// Set CR to always fail.
+	oldMode := fbo.cr.getFailModeForTest()
+	fbo.cr.setFailModeForTest(alwaysFailCR)
+	defer func() { fbo.cr.setFailModeForTest(oldMode) }()
```

Comment:
It doesn't feel like the locking is doing much good on `failModeForTest` if we don't take the lock during this entire time - someone else could set the fail mode to fail always, then we set it to fail always and take their fail always as the thing to put back. Then they're done and come turn it off, and we come turn it back on. There isn't any other code accessing this, but it still feels misleading to have the locks like this if they're not effective.

WDYT?

Reply:
The lock is just intended to prevent test data races on the access of that specific variable.  If we happen to have simultaneous get and set calls (if CR is somehow running the background already when we call this, for example).